samcli/hook_packages/terraform/hooks/prepare.py

```diff
@@ -986,6 +1140,7 @@ def _get_s3_object_hash(bucket: str, key: str) -> str:
     md5 = hashlib.md5()
     md5.update(bucket.encode())
     md5.update(key.encode())
+    # TODO: Hash version if it exists in addition to key and bucket
```

Comment:
are you going to add it in a separate PR ?

Reply:
Yes I'll raise a separate PR after this one to keep the size of this one smaller. I also created an internal ticket to track this item.